pkg/sql/schemachanger/scplan/plan.go

```diff
@@ -165,11 +165,7 @@ func buildStages(init scpb.State, g *scgraph.Graph, params Params) []Stage {
 				return nodeRanks[edges[i].To()] > nodeRanks[edges[j].To()]
 			})
 
-		next := scpb.State{
-			Nodes:         append(cur.Nodes[0:0], cur.Nodes...),
-			Statements:    append(cur.Statements[0:0], cur.Statements...),
```

Comment:
I wonder how much time and money will be lost over time by programmers confusing `[0:0]` with `[:0:0]` in go. It's such an easy bug to make. It's crazy.

Reply:
Fixed it to not use that subtle pattern. What I did is more efficient anyway. This is a thing where generics will help. In go 1.18, this will be `slices.Clone(cur.Nodes)`

https://github.com/golang/exp/blob/ba1b3c35638383b1076303068d7b583b5225b880/slices/slices.go#L156-L158